source/git/source.go

```diff
@@ -635,6 +652,16 @@ func (gs *gitSourceHandler) Snapshot(ctx context.Context, g session.Group) (out
 	}
 
 	git = git.New(gitutil.WithWorkTree(checkoutDir), gitutil.WithGitDir(gitDir))
+	if gs.src.CommitHash != "" {
```

Comment:
Why is this happening after we have already completed the checkout instead of before?

Reply:
Just for verification, as we may potentially have an unexpected code path (in future)